dbt/adapters/snowflake/impl.py

```diff
@@ -100,7 +102,8 @@ def post_model_hook(self, config: Mapping[str, Any], context: Optional[str]) ->
 
     def list_schemas(self, database: str) -> List[str]:
         try:
-            results = self.execute_macro(LIST_SCHEMAS_MACRO_NAME, kwargs={"database": database})
+            compilation_db = CompilationDBReplacer.customer_db_to_compilation_db(database)
+            results = self.execute_macro(LIST_SCHEMAS_MACRO_NAME, kwargs={"database": compilation_db})
```

Comment:
hmm the name of the database in the request should already be replaced by the code you added in the snowflake adapter no..? why do you need to replace it here too? (same question for all changes in this file)

Reply:
All these functions are called directly from 'dbt' package, in addition to calling functions in connections.py